src/zope/security/checker.py

```diff
@@ -770,6 +770,27 @@ def f(): #pragma NO COVER
     _default_checkers[type({}.iterkeys())] = _iteratorChecker
     _default_checkers[type({}.itervalues())] = _iteratorChecker
 
+def _fixup_dictlike(dict_type):
+    empty_dict = dict_type()
+    populated_dict = dict_type({1: 2})
+    for dictlike in (empty_dict, populated_dict):
+        for attr in ('__iter__', 'keys', 'items', 'values'):
+            obj = getattr(dictlike, attr)()
+            o_type = type(obj)
+            if o_type not in _default_checkers:
+                _default_checkers[o_type] = _iteratorChecker
+
+def _fixup_odict():
+    # OrderedDicts have three different implementations: Python 2 (pure
+    # python, returns generators and lists), Python <=3.4 (pure Python,
+    # uses view classes) and CPython 3.5+ (implemented in C). These should
+    # all be iterable.
+    from collections import OrderedDict
+    _fixup_dictlike(OrderedDict)
+
+_fixup_odict()
+del _fixup_odict
```

Comment:
Oh. Well, maybe `_fixup` is a bad name then. This is just making sure that `_default_checkers` is fully and correctly populated. I don't see that as a side-effect, since that's a substantial part of what this module does.